### PR TITLE
BA 1936: create chat groups

### PR DESCRIPTION
**Pull request by Ronan-Fernandes**
## Release Notes

- **New Features**
  - Introduced a `CreateGroup` component for creating group chat rooms with form validation and image uploads.
  - Added a `ProfileCard` component to display user profiles and manage group membership actions.
  - Implemented a `CircledAvatar` component for displaying user avatars in a circular format.
  - Added a `FileUploadButton` component for handling file uploads with validation.
  - Introduced a `ConnectionsList` component for managing and displaying a list of user connections.
  - Enhanced `CreateChatRoomList` with new functionality for group chat creation.

- **Improvements**
  - Enhanced the `MessagesGroup` component to simplify rendering logic based on group status.
  - Updated the `ChatRoom` GraphQL fragment to include an `isGroup` field for better chat room categorization.
  - Streamlined the `getSubtitle` function to directly check the `isGroup` property.
  - Improved profile data handling with new structured types for profiles.

- **Bug Fixes**
  - Fixed height bugs associated with Virtuoso lists.

- **Documentation**
  - Added detailed documentation for the `FileUploadButton` component in Storybook.
  - Updated the documentation for the `CreateGroup` component to reflect its new functionalities.

- **Chores**
  - Removed unused constants and type declarations related to profile handling.

### CODE REVIEW

File: packages/components/modules/messages/ChatRoomsList/index.tsx [hunk 1]
```diff
@@ -100,10 +100,10 @@ const ChatRoomsList: FC<ChatRoomsListProps> = ({
     )
   }
 
-  const renderTabLabel = (tabValue: ChatTabValues) => (
+  const renderTabLabel = (tabValue: keyof typeof CHAT_TAB_VALUES) => (
```
Comment (anicioalexandre) [inline packages/components/modules/messages/ChatRoomsList/index.tsx]: isn't it the same thing as ChatTabValues?

File: packages/components/modules/messages/CreateGroup/index.tsx [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+        connections: [
+          // TODO: add filter handling (for now we can default 'unreadMessages' to false)
+          ConnectionHandler.getConnectionID(currentProfile?.id as string, 'roomsList_chatRooms', {
+            unreadMessages: false,
+          }),
+        ],
+      },
+      uploadables,
+      onCompleted: (response) => {
+        setChatRoom({ id: response?.chatRoomCreate?.room?.node?.id })
```
Comment (tsl-ps2) [inline packages/components/modules/messages/CreateGroup/index.tsx]: Do we need some error handling here (as for example in apps/web/app/(with-navigation)/user/settings/AccountSettings/AccountProfile/index.tsx)?

File: packages/components/modules/messages/CreateGroup/index.tsx [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+          <Box
+            role="list"
+            aria-label="Available contacts"
+            sx={{
+              padding: theme.spacing(2),
+              borderBottom: `1px solid ${theme.palette.divider}`,
+            }}
+          >
+            <Typography variant="subtitle2" color="text.primary">
+              contacts
```
Comment (tsl-ps2) [inline packages/components/modules/messages/CreateGroup/index.tsx]: "Contacts" (capital C)?

File: packages/components/modules/messages/CreateGroup/styled.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    padding: `${theme.spacing(2)} ${theme.spacing(1.5)} 0`,
+  },
+}))
+
+export const HeaderContainer = styled(Box)(({ theme }) => ({
+  display: 'grid',
+  gridTemplateColumns: '24px auto 24px',
+  width: '100%',
+  padding: theme.spacing(2),
+  borderBottom: `1px solid ${theme.palette.divider}`,
```
Comment (anicioalexandre) [inline packages/components/modules/messages/CreateGroup/styled.tsx:17]: let's add a `alignItems: "center"`

File: packages/components/modules/messages/CreateGroup/types.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import { SearchbarProps } from '@baseapp-frontend/design-system'
+
+import { VirtuosoProps } from 'react-virtuoso'
+
+import { ChatRoomsQuery$data } from '../../../__generated__/ChatRoomsQuery.graphql'
+import { ProfileCardProps } from './ProfileCard/types'
+
+export interface CreateGroupProps extends PropsWithChildren {
+  allProfilesRef: ChatRoomsQuery$data
+  remotePatternsHostName?: string
```
Comment (anicioalexandre) [inline packages/components/modules/messages/CreateGroup/types.ts]: I dont think we need this as a prop, we probably wont be changing it

File: packages/components/modules/messages/CreateGroup/index.tsx [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+
+  const { currentProfile } = useCurrentProfile()
+
+  const [commit, isMutationInFlight] = useCreateChatRoomMutation()
+  const { setChatRoom } = useChatRoom()
+
+  const onSubmit = handleSubmit((data: any) => {
+    const dirtyValues = filterDirtyValues({ values: data, dirtyFields })
+    const { title, participants, image } = data
+    const participantsIds = participants.map((member: any) => member.id)
```
Comment (anicioalexandre) [inline packages/components/modules/messages/CreateGroup/index.tsx]: we could better type that
```suggestion
  const onSubmit = handleSubmit((data: CreatGroupUpload) => {
    const dirtyValues = filterDirtyValues({ values: data, dirtyFields })
    const { title, participants, image } = data
    const participantsIds = participants?.map((member) => member!.id) || []
```

File: packages/components/modules/messages/CreateGroup/index.tsx [hunk 7]
```diff
@@ -0,0 +1,356 @@
...
+  }
+
+  const handleSearchClear = () => {
+    startTransition(() => {
+      searchReset()
+      refetch({ q: '' })
+    })
+  }
+
+  const participants = watch(FORM_VALUE.participants) as any[]
```
Comment (anicioalexandre) [inline packages/components/modules/messages/CreateGroup/index.tsx]:
```suggestion
  const participants = watch(FORM_VALUE.participants) as ProfileNode[]

```

File: packages/components/modules/messages/CreateGroup/index.tsx [hunk 8]
```diff
@@ -0,0 +1,356 @@
...
+
+  const profiles = useMemo(
```
Comment (anicioalexandre) [inline packages/components/modules/messages/CreateGroup/index.tsx:155]:
```suggestion
  const profiles = useMemo(
    () =>
      allProfiles?.edges
        .filter(
          (edge) =>
            edge?.node &&
            edge?.node.id !== currentProfile?.id &&
            !participants?.some((member) => member?.id === edge?.node?.id),
        )
        .map((edge) => edge?.node) || [],
    [allProfiles, participants, currentProfile?.id],
  )
```

File: packages/components/modules/messages/CreateGroup/index.tsx [hunk 9]
```diff
@@ -0,0 +1,356 @@
...
+          </UploadProfileContainer>
+          <TextField
+            label="Group Name"
+            control={control}
+            name={FORM_VALUE.title}
+            className="h-[min-content]"
+          />
+        </Box>
+      </Box>
+      <Box height="100%" width="100%">
```
Comment (anicioalexandre) [inline packages/components/modules/messages/CreateGroup/index.tsx]: let's add an overflow: "auto here too, so we avoid losing the Contacts section when the Member one gets bigger

File: packages/components/modules/messages/CreateGroup/index.tsx (hunk 2 above)
Comment (Ronan-Fernandes) [reply packages/components/modules/messages/CreateGroup/index.tsx]: Nice catch I'll update it, thanks

**Review by anicioalexandre — APPROVED** [review]

**Pull request merged**